pom.xml

```diff
@@ -610,6 +635,24 @@ THE SOFTWARE.
                 </goals>
               </execution>
             </executions>
+            <configuration>
+              <keyname>${env.GPG_KEYNAME}</keyname>
+              <passphrase>${env.GPG_PASSPHRASE}</passphrase>
+              <!-- This is necessary for gpg to not try to use the pinentry programs -->
+              <gpgArguments>
+                  <arg>--pinentry-mode</arg>
+                  <arg>loopback</arg>
+              </gpgArguments>
+            </configuration>
+          </plugin>
+          <plugin>
+            <artifactId>maven-jarsigner-plugin</artifactId>
+            <configuration>
```

Comment:
Such implementation will not work on my machine with a hardware codesigning key. But I;d guess it will be fine for @kohsuke, so why not.

I suggest adding some flags to enforce the code signing verification, e.g. sample from Remoting

```
<configuration>
                  <processMainArtifact>true</processMainArtifact>
                  <processAttachedArtifacts>false</processAttachedArtifacts>
                  <certs>true</certs>
                  <errorWhenNotSigned>true</errorWhenNotSigned>
                  <arguments>-strict</arguments> <!--otherwise certificate chains will be ignored-->
                  <alias>${hudson.sign.alias}</alias>
                  <storepass>${hudson.sign.storepass}</storepass>
                  <keystore>${hudson.sign.keystore}</keystore>
                  <storetype>${hudson.sign.storetype}</storetype>
                  <providerClass>${hudson.sign.providerClass}</providerClass>
                  <providerArg>${hudson.sign.providerArg}</providerArg>
                </configuration>
```

Reply:
```suggestion
            <configuration>
              <certs>true</certs>
              <errorWhenNotSigned>true</errorWhenNotSigned>
              <arguments>-strict</arguments> <!--otherwise certificate chains will be ignored-->
```

Reply:
I think the goal is to be able to use a software signing certificate, obtained on behalf of the CDF. And for Kohsuke to also be able to use it, at least for the interim till we get everything working correctly. If it doesn't work for a hardware-based codesigning key, that's probably not a concern.

Reply:
+1 on enforcing the code signing when this profile is used.

Reply:
https://github.com/jenkins-infra/release/blob/master/utils/release.sh#L221-L219